src/main/java/factionsplusplus/services/FactionService.java

```diff
@@ -121,9 +122,10 @@ public int getCumulativePowerLevel(Faction faction) {
     public int getMaximumCumulativePowerLevel(Faction faction) {     // get max power without vassal contribution
         int maxPower = 0;
 
-        for (UUID playerUUID : faction.getMemberList()) {
+        for (GroupMember member : faction.getMembers()) {
+            UUID playerUuid = member.getId();
             try {
-                maxPower += this.playerService.getMaxPower(playerUUID);
+                maxPower += this.playerService.getMaxPower(playerUuid);
```

Comment:
the variable is single use, perhaps just pass member.getId() here?

Reply:
Are you suggesting to loop through the member values of the map over the keySet?